src/event-stream/tokens-contract-handler.ts

```diff
@@ -626,27 +649,90 @@ class TokensContractHandler {
   /**
    * Make readonly contract call
    */
-  private async makeReadOnlyContractCall(
+  async makeReadOnlyContractCall(
+    functionName: string,
+    functionArgs: ClarityValue[]
+  ): Promise<ReadOnlyContractCallResponse> {
+    const url = this.stacksNetwork.getReadOnlyFunctionCallApiUrl(
+      this.contractAddress,
+      this.contractName,
+      functionName
+    );
+    const args = functionArgs.map(arg => cvToHex(arg));
+
+    const body = JSON.stringify({
+      sender: this.address,
+      arguments: args,
+    });
+
+    let response: Response;
+
+    try {
+      response = await fetchPrivate(url, {
+        method: 'POST',
+        body,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    } catch (error: any) {
+      // In case there is a network error during fetch.
+      console.log(`Error calling read only contract function value: ${error.message}`);
+      return { found: false, retriable: true };
+    }
+
+    if (!response.ok) {
+      if (response.status >= 500) {
+        return { found: false, retriable: true };
+      }
+      const msg = await response.text().catch(() => '');
+      return {
+        found: false,
+        retriable: false,
+        error: `Error calling read-only function. Response ${response.status}: ${response.statusText}. Attempted to fetch ${url} and failed with the message: "${msg}"`,
+      };
+    }
+    try {
+      const value: ReadOnlyFunctionResponse = await response.json();
+      const parseValue = parseReadOnlyResponse(value);
+      if (parseValue.type === ClarityType.ResponseErr) {
+        return {
+          found: false,
+          retriable: false,
+          error: `Function execution failed with error: ${parseValue.value}`,
+        };
+      }
+      return { found: true, value: parseValue };
+    } catch (error: any) {
+      // In case there was a connection was interrupted
+      console.log(`Error parsing read only contract function response: ${error.message}`);
```

Comment:
These should use the `logger` instance and match the convention used in the rest of the module, e.g.
```js
logger.warn(
  `[token-metadata] ...<information about the error and contract ID>...`,
  error
);
```